rosbag2/src/rosbag2/writer.cpp

```diff
@@ -76,6 +89,13 @@ void Writer::create_topic(const TopicMetadata & topic_with_type)
     converter_->add_topic(topic_with_type.name, topic_with_type.type);
   }
 
+  if (topics_.find(topic_with_type.name) == topics_.end()) {
+    rosbag2_storage::TopicInformation info{};
+    info.topic_metadata = topic_with_type;
+
+    topics_.insert({topic_with_type.name, info});
```

Comment:
We should check `insert` return value and assert or throw if the topic already exist